### ISSUE TITLE
Improve the error handler when duplicate key exists

### ISSUE THREAD

**Issue by zhenggu** [antw/iniparse#22]
when there is two duplicate keys with the same name, and I call IniParse.open, it doesn't report any error, but when call the key function of each element, it crashes with the error information:

```
in `block in <main>': undefined method `key' for #<Array:0x00007fc2e0ac3a28> (NoMethodError)
	from /Library/Ruby/Gems/2.3.0/gems/iniparse-1.4.4/lib/iniparse/line_collection.rb:62:in `block in each'
	from /Library/Ruby/Gems/2.3.0/gems/iniparse-1.4.4/lib/iniparse/line_collection.rb:60:in `each'
	from /Library/Ruby/Gems/2.3.0/gems/iniparse-1.4.4/lib/iniparse/line_collection.rb:60:in `each'
	from /Library/Ruby/Gems/2.3.0/gems/iniparse-1.4.4/lib/iniparse/lines.rb:128:in `each'
	from ./test.rb:6:in `map'
	from ./test.rb:6:in `<main>'
```

It make the users confused what happened, so there could be two improvements:
1. Throw an error when I call IniParse.open
2. Make the error message more reasonable

Test code:

```
#!/usr/bin/ruby
require 'iniparse'
ini = IniParse.open 'test.ini'
ini['test'].map {|a| a.key }
```

Test ini file:

```
[test]
a = 1
a = 2
```

**Comment by antw** [comment]
Thanks for the report. Fixed in v1.5.0 (just published).

```ini
[test]
a = 1
a = 2
b = 3
```

```ruby
ini['test'].each do |a|
  # 1) yields a = 1
  # 2) yields a = 2
  # 3) yields b = 3
  # -) stops
end

ini['test'].map { |a| [a.key, a.value] }
# => [["a", 1], ["a", 2], ["b", 3]]
```

**Comment by zhenggu** [comment]
Thanks very much
